icu4c/source/i18n/unicode/simplenumberformatter.h

```diff
@@ -0,0 +1,331 @@
+    USimpleNumberSign fSign = UNUM_SIMPLE_NUMBER_NO_SIGN;
+
+    friend class SimpleNumberFormatter;
+};
+
+
+/**
+ * A special NumberFormatter focused on smaller binary size and memory use.
+ * 
+ * SimpleNumberFormatter is capable of basic number formatting, including grouping separators,
+ * sign display, and rounding. It is not capable of currencies, compact notation, or units.
+ *
+ * This class is immutable and not intended for public subclassing.
+ *
+ * @draft ICU 73
+ */
+class U_I18N_API SimpleNumberFormatter : public UMemory {
+  public:
+    /**
+     * Creates a new SimpleNumberFormatter with all locale defaults.
+     *
+     * @draft ICU 73
+     */
+    static SimpleNumberFormatter forLocale(
+        const icu::Locale &locale,
+        UErrorCode &status);
+
+    /**
+     * Creates a new SimpleNumberFormatter, overriding the grouping strategy.
+     *
+     * @draft ICU 73
+     */
+    static SimpleNumberFormatter forLocaleAndGroupingStrategy(
+        const icu::Locale &locale,
+        UNumberGroupingStrategy groupingStrategy,
+        UErrorCode &status);
+
+    /**
+     * Creates a new SimpleNumberFormatter, overriding the grouping strategy and symbols.
+     *
+     * IMPORTANT: For efficiency, this function borrows the symbols. The symbols MUST remain valid
+     * for the lifetime of the SimpleNumberFormatter.
+     *
+     * @draft ICU 73
+     */
+    static SimpleNumberFormatter forLocaleAndSymbolsAndGroupingStrategy(
+        const icu::Locale &locale,
+        const DecimalFormatSymbols &symbols,
+        UNumberGroupingStrategy groupingStrategy,
+        UErrorCode &status);
+
+    /**
+     * Formats a value using this SimpleNumberFormatter.
+     *
+     * The SimpleNumber argument is "consumed". A new SimpleNumber object should be created for
+     * every formatting operation.
+     *
+     * @draft ICU 73
+     */
+    FormattedNumber format(SimpleNumber value, UErrorCode &status) const;
+
+    /**
+     * Formats an integer using this SimpleNumberFormatter.
+     *
+     * For more control over the formatting, use SimpleNumber.
+     *
+     * @draft ICU 73
+     */
+    FormattedNumber formatInt64(int64_t value, UErrorCode &status) const {
+        return format(SimpleNumber::forInt64(value, status), status);
+    }
+
+#ifndef U_HIDE_INTERNAL_API
+    /**
+     * Run the formatter with the internal types.
+     * @internal
+     */
+    void formatImpl(impl::UFormattedNumberData* data, USimpleNumberSign sign, UErrorCode& status) const;
+#endif // U_HIDE_INTERNAL_API
+
+    /**
+     * Destruct this SimpleNumberFormatter, cleaning up any memory it might own.
+     *
+     * @draft ICU 73
+     */
+    ~SimpleNumberFormatter() {
+        cleanup();
+    }
+
+    /**
+     * Creates a shell, initialized but non-functional SimpleNumberFormatter.
+     *
+     * @draft ICU 73
+     */
+    SimpleNumberFormatter() = default;
+
+    // FIXME: Make sure these move constructors are well tested
```

Comment:
reminder...

Reply:
done (I wrote the test but forgot to remove the reminder)